clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -79,15 +101,17 @@ public void subscribe(Pattern pattern, final ConsumerRebalanceListener listener)
     }
 
     @Override
-    public synchronized void subscribe(List<String> topics, final ConsumerRebalanceListener listener) {
+    public void subscribe(List<String> topics, final ConsumerRebalanceListener listener) {
         ensureNotClosed();
         this.subscriptions.subscribe(topics, SubscriptionState.wrapListener(this, listener));
     }
 
     @Override
-    public synchronized void assign(List<TopicPartition> partitions) {
+    public void assign(List<TopicPartition> partitions) {
         ensureNotClosed();
         this.subscriptions.assign(partitions);
+        for (TopicPartition tp : partitions)
+            this.subscriptions.needOffsetReset(tp);
```

Comment:
The "real" consumer doesn't reset offsets when assigning partitions, why do we need to mock it?